### PR TITLE
provide pushMetric with objects

### PR DESCRIPTION
**Pull request by ploki**
Provide pushMetric with objects on the methods with storage side effects

The purpose of this PR is to permit other services accessing the same data set to keep their states consistent

### CODE REVIEW

**Comment by ploki** [comment]
@ironman-machine SCALITY_UTAPI_BRANCH=ft/publish-side-effects

**Comment by ploki** [comment]
@ironman-machine try

**Comment by ploki** [comment]
@ironman-machine try

File: lib/api/multiObjectDelete.js [hunk 1]
```diff
@@ -483,6 +483,7 @@ function multiObjectDelete(authInfo, request, log, callback) {
         pushMetric('multiObjectDelete', log, {
             authInfo,
             bucket: bucketName,
+            keys: successfullyDeleted,
```
Comment (electrachong) [inline lib/api/multiObjectDelete.js]: it's ok if the array item is an object containing `{ versionId, key }` instead of being an array of strings?

Comment (ploki) [reply lib/api/multiObjectDelete.js]: No it isn't. Is the ```successfullyDeleted``` array always composed of objects of this kind?

Comment (electrachong) [reply lib/api/multiObjectDelete.js]: yes, at least in any release with versioning implemented (>= 7.0)

**Comment by ploki** [comment]
@ironman-machine try

**Comment by ploki** [comment]
@ironman-machine try

**Comment by ploki** [comment]
again @ironman-machine try

File: lib/api/multiObjectDelete.js [hunk 2]
```diff
@@ -480,9 +480,12 @@ function multiObjectDelete(authInfo, request, log, callback) {
         }
         const xml = _formatXML(quietSetting, errorResults,
             successfullyDeleted);
+        const deletedKeys = [];
+        successfullyDeleted.map(item => deletedKeys.push(item.key));
```
Comment (electrachong) [inline lib/api/multiObjectDelete.js]: you can do this, no need to create a separate array outside:
`const deletedKeys = successfullyDeleted.map(item => item.key);`

**Comment by electrachong** [comment]
@ironman-machine r+

**Comment by electrachong** [comment]
@ironman-machine 
SCALITY_UTAPI_BRANCH=
r+

**Comment by electrachong** [comment]
@ironman-machine r+

**Pull request merged**